hadoop-master1/files/etc/ranger/startup.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -xe
+
+function setup_config() {
+    cp /etc/ranger/conf/* ${RANGER_HOME}/
+    sed -i 's/__REPLACE_MYSQL_JDBC_VERSION__/'"${MYSQL_JDBC_VERSION}"'/g' "${RANGER_HOME}/install.properties"
+}
+
+function setup_db() {
+    db_exists=$(/opt/openjdk-8/bin/java -cp /opt/ranger/mysql-connector-j-${MYSQL_JDBC_VERSION}.jar:/opt/ranger-${RANGER_VERSION}-admin/jisql/lib/* org.apache.util.sql.Jisql -driver mysqlconj -cstring jdbc:mysql://mysql:3306/mysql -u root -p 'root' -noheader -trim -c \; -query "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = 'ranger';")
+    if [ -z "$db_exists" ]; then
+        /opt/openjdk-8/bin/java -cp /opt/ranger/mysql-connector-j-${MYSQL_JDBC_VERSION}.jar:/opt/ranger-${RANGER_VERSION}-admin/jisql/lib/* org.apache.util.sql.Jisql -driver mysqlconj -cstring jdbc:mysql://mysql:3306/mysql -u root -p 'root' -noheader -trim -c \; -query "CREATE DATABASE ranger;"
```

Comment:
simply put the init sql at `mysql/files/docker-entrypoint-initdb.d`

Reply:
ok